src/logworker.cpp

```diff
@@ -120,6 +120,20 @@ namespace g3 {
       return std::unique_ptr<LogWorker>(new LogWorker);
    }
 
+   void LogWorker::removeWrappedSink(std::shared_ptr<g3::internal::SinkWrapper> sink) {
+      auto bg_removesink_call = [this, sink] {
+         for (auto it = _impl._sinks.begin(); it != _impl._sinks.end(); ) {
+            if ((*it).get() == sink.get()) {
+               it = _impl._sinks.erase(it);
+            } else {
+               ++it;
+            }
+         }
+      };
+      auto token_done = g3::spawn_task(bg_removesink_call, _impl._bg.get());
```

Comment:
I'm not sure if we have to worry about nullptrs. Do you see any cases where a sink handle can be removed and then the sink handle is re-used again but now resulting in that the lock to the sink handle returns nullptr?